src/main/java/uk/gov/ons/ctp/response/sample/service/SampleSummaryActivationService.java

```diff
@@ -0,0 +1,135 @@
+   * and enrich the sample unit by gathering any extra data required for the second step of the
+   * activation process.
+   *
+   * @param sampleSummaryActivation an object containing all the details required for activation
+   */
+  private void validateAndEnrich(SampleSummaryActivationDTO sampleSummaryActivation)
+      throws SampleSummaryActivationException {
+    UUID sampleSummaryId = sampleSummaryActivation.getSampleSummaryId();
+    UUID surveyId = sampleSummaryActivation.getSurveyId();
+    UUID collectionExerciseId = sampleSummaryActivation.getCollectionExerciseId();
+
+    LOG.debug(
+        "about to enrich sample summary",
+        kv("sampleSummaryId", sampleSummaryId),
+        kv("surveyId", surveyId),
+        kv("collectionExerciseId", collectionExerciseId));
+
+    try {
+      boolean validated =
+          sampleSummaryEnrichmentService.enrich(surveyId, sampleSummaryId, collectionExerciseId);
+      LOG.debug(
+          "Enriched sample summary",
+          kv("sampleSummaryId", sampleSummaryId),
+          kv("surveyId", surveyId),
+          kv("collectionExerciseId", collectionExerciseId),
+          kv("validated", validated));
+      if (validated) {
+        LOG.info("Sample summary successfully enriched", kv("sampleSummaryId", sampleSummaryId));
+      } else {
+        LOG.error("Validation and enrichment failed", kv("sampleSummaryId", sampleSummaryId));
+        throw new SampleSummaryActivationException();
+      }
+    } catch (UnknownSampleSummaryException e) {
+      LOG.error("unknown sample summary id", kv("sampleSummaryId", sampleSummaryId), e);
+      sendEnrichStatusToCollectionExercise(collectionExerciseId, false);
+      throw new SampleSummaryActivationException();
+    } catch (SampleSummaryActivationException e) {
+      LOG.error(
+          "Something went wrong activating sample summary",
+          kv("sampleSummaryId", sampleSummaryId),
+          e);
+      sendEnrichStatusToCollectionExercise(collectionExerciseId, false);
+      throw new SampleSummaryActivationException();
+    }
+  }
+
+  /**
+   * Executes the second step of the activation process. This will send each sample unit in a sample
+   * summary to the case service to create cases for each one (which are used to track the progress
+   * of the survey submission).
+   *
+   * @param sampleSummaryActivation an object containing all the details required for activation
+   */
+  private void distribute(SampleSummaryActivationDTO sampleSummaryActivation)
+      throws SampleSummaryActivationException {
+    try {
+      sampleSummaryDistributionService.distribute(sampleSummaryActivation.getSampleSummaryId());
+    } catch (NoSampleUnitsInSampleSummaryException | UnknownSampleSummaryException e) {
+      LOG.error(
+          "something went wrong during distribution sample units",
+          kv("sampleSummaryActivation", sampleSummaryActivation));
```

Comment:
again I think it might be worth logging the individual bits of the sampleSummaryActivation